runtime/resolvers/resolvers_test.go

```diff
@@ -0,0 +1,124 @@
+package resolvers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"maps"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/maxatome/go-testdeep/td"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/testruntime"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+type Project struct {
+	Sources    map[string]yaml.Node
+	Models     map[string]yaml.Node
+	Dashboards map[string]yaml.Node
+	APIs       map[string]yaml.Node
+}
+type Resolvers struct {
+	Project    Project
+	Connectors map[string]testruntime.InstanceOptions
+	Tests      map[string]*Test
+}
+
+type Test struct {
+	Resolver string
+	Options  runtime.ResolveOptions
+	Result   []map[string]any
+}
```

Comment:
Instead of turning internal types like `testruntime.InstanceOptions` and `runtime.ResolveOptions` into YAML serializable types, I would recommend explicitly adding the relevant fields here and then creating those internal types separately in the test initializer